site/content/en/guides/producer/functions/ts/_index.md

```diff
@@ -4,8 +4,103 @@ linkTitle: "TypeScript Function SDK"
 weight: 5
 type: docs
 description: >
-   Writing exec and container functions in TypeScript.
+   Writing functions in TypeScript.
 ---
 
-See [kpt-functions-sdk](https://googlecontainertools.github.io/kpt-functions-sdk/) for
-instructions on writing functions as TypeScript programs.
+We provide an opinionated Typescript SDK for implementing KPT Functions. This provides various
+advantages:
+
+- **General-purpose language:** Domain-Specific Languages begin their life with a reasonable
+  feature set, but often grow over time. They bloat in order to accommodate the tremendous variety
+  of customer use cases. Rather than follow this same course, KPT functions employ a true,
+  general-purpose programming language that provides:
+  - Proper abstractions and language features
+  - A extensive ecosystem of tooling (e.g. IDE support)
+  - A comprehensive catalog of well-supported libraries
+  - Robust community support and detailed documentation
+- **Type-safety:** Kubernetes configuration is typed, and its schema is defined using the OpenAPI spec.
+  Typescript has a sophisticated type system that accommodates the complexity of Kubernetes resources.
+  The SDK enables generating Typescript classes for core and CRD types, providing safe and easy
+  interaction with Kubernetes objects.
+- **Batteries-included:** The SDK provides a simple, powerful API for querying and manipulating configuration
+  files. It provides the scaffolding required to develop, build, test, and publish functions,
+  allowing you to focus on implementing your business-logic.
+
+## Developer Quickstart
+
+This quickstart will get you started developing a KPT Function with the TypeScript SDK,
+using an existing Hello World package.
+
+### Prerequisites
+
+#### System Requirements
+
+Currently supported platforms: amd64 Linux/Mac
+
+#### Setting Up Your Local Environment
+
+- Install [node][download-node]
+- Install [docker][install-docker]
+- Install [kpt][download-kpt] and add it to \$PATH
+
+### Hello World Package
+
+1. Get the `hello-world` package:
+
+   ```sh
+   git clone --depth 1 https://github.com/GoogleContainerTools/kpt-functions-sdk.git
+   ```
+
+   All subsequent commands are run from the `hello-world` directory:
+
+   ```sh
+   cd kpt-functions-sdk/ts/hello-world
+   ```
+
+1. Install all dependencies:
+
+   ```sh
+   npm install
+   ```
+
+1. Run the following in a separate terminal to continuously build your function as you make changes:
+
+   ```sh
+   npm run watch
+   ```
+
+1. Run the `label_namespace` function:
```

Comment:
In this file we should include the function inline (I think there's a way to link instead of copy code). This gives the reader an idea of what TS functions look like and is consistent with other languages.

Reply:
Linked to the function.